test/e2e/framework/pod/wait.go

```diff
@@ -275,8 +275,8 @@ func WaitForPodCondition(c clientset.Interface, ns, podName, conditionDesc strin
 	if err == nil {
 		return nil
 	}
-	if IsTimeout(err) {
-		e2elog.Logf("Timed out while waiting for pod %s to be %s. Last observed as: %s",
+	if IsTimeout(err) && lastPod != nil {
+		return TimeoutError("timed out while waiting for pod %s to be %s: last observed as:\n%s",
```

Comment:
How does our test infrastructure aggregate failures? If anything is looking for identical failure messages, this will no longer match.

Reply:
Will no longer match because the dumped pod state will always be different, right?

The first line will still be the same, so searching for that should still work. I don't know whether that is going to be enough. Let me ask on #sig-testing.

Reply:
It looks like failure text differences are already handled:

https://storage.googleapis.com/k8s-triage/index.html?text=timed%20out%20while%20waiting

```
351 test failures (44 today) look like [link](https://storage.googleapis.com/k8s-triage/index.html?text=timed%20out%20while%20waiting#32d66067bb5fe4ce2e3e)[search github](https://github.com/search?type=Issues&q=org:kubernetes%2032d66067bb5fe4ce2e3e)[file bug](https://storage.googleapis.com/k8s-triage/index.html?text=timed%20out%20while%20waiting#)storage

test/e2e/storage/nfs_persistent_volume-disruptive.go:153
May  5 11:08:28.112: Unexpected error:
    <*errors.errorString | 0xc00654af50>: {
        s: "pod \"pvc-tester-8c6qg\" is not Running: timed out while waiting for pod disruptive-pv-142/pvc-tester-8c6qg to be running",
    }
    pod "pvc-tester-8c6qg" is not Running: timed out while waiting for pod disruptive-pv-142/pvc-tester-8c6qg to be running
occurred
test/e2e/storage/nfs_persistent_volume-disruptive.go:185
```

Note that pod name and namespace already vary in each failure instance.

Reply:
Ben explained [on Slack](https://kubernetes.slack.com/archives/C09QZ4DQB/p1652423870096959?thread_ts=1652423415.090849&cid=C09QZ4DQB) that this particular example works because varying strings like the 142 suffix get removed via regular expressions. Then remaining content must have <10% editing difference.

So including the pod could be a real problem. It could also help if the additional information let's the tool distinguish between different root causes for a failure, but I am a bit pessimistic about that.

Let's drop this part?